### ISSUE TITLE
mis-reported duplicates

### ISSUE THREAD

**Issue by jtpereyda** [0vercl0k/rp#29]
Syscall gadgets can be misreported as duplicates. Example output (first run with --unique, second without):

```
> ./rp-lin-x64 --unique -r2 -f test.elf
Trying to open 'test.elf'..
Loading ELF information..
FileFormat: Elf, Arch: x64

Wait a few seconds, rp++ is looking for gadgets (2 threads max)..
A total of 7 gadgets found.
You decided to keep only the unique ones, 3 unique gadgets found.
0x00401004: ret  ;  \xc3 (1 found)
0x00401000: syscall  ;  \x45\xf0\x0f\x05 (3 found)
0x00401000: syscall  ; ret  ;  \x45\xf0\x0f\x05\xc3 (3 found)

> ./rp-lin-x64 -r2 -f test.elf
Trying to open 'test.elf'..
Loading ELF information..
FileFormat: Elf, Arch: x64

Wait a few seconds, rp++ is looking for gadgets (2 threads max)..
A total of 7 gadgets found.
0x00401000: syscall  ;  \x45\xf0\x0f\x05 (1 found)
0x00401001: syscall  ;  \xf0\x0f\x05 (1 found)
0x00401002: syscall  ;  \x0f\x05 (1 found)
0x00401004: ret  ;  \xc3 (1 found)
0x00401000: syscall  ; ret  ;  \x45\xf0\x0f\x05\xc3 (1 found)
0x00401001: syscall  ; ret  ;  \xf0\x0f\x05\xc3 (1 found)
0x00401002: syscall  ; ret  ;  \x0f\x05\xc3 (1 found)
```

However, these syscalls are not all the same. See these disassembly results from pwntools:

```
> ipython3
Python 3.8.10 (default, Sep 28 2021, 16:10:42) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.13.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import pwn                                                                                      

In [2]: pwn.disasm(b'\x0f\x05')                                                                         
Out[2]: '   0:   0f 05                   syscall'

In [3]: pwn.disasm(b'\xf0\x0f\x05')                                                                     
Out[3]: '   0:   f0 0f 05                lock syscall'

In [4]: pwn.disasm(b'\x45\xf0\x0f\x05')                                                                 
Out[4]: '   0:   45                      inc    ebp\n   1:   f0 0f 05                lock syscall'
```

And in practice, only the smaller gadget at 0x401002 works for me as a syscall.

I suspect the disassembly just needs to be tweaked somehow to fix.

**Comment by 0vercl0k** [comment]
Thanks for opening a detailed issue!

This definitely looks related to the disassembler - I'll take a look when I work on the anniversary update.

Cheers

**Comment by 0vercl0k** [comment]
All right - as promised I've been working on this clean-up and BeaEngine got an update as well which seems to be fixing your issues:
```
> rp-win-x64.exe -f foo.bin --raw=x64 --rop=1
0x0: lock pop rdi ; ret ; \xf0\x5f\xc3 (1 found)
0x2: ret ; \xc3 (1 found)
0x1: pop rdi ; ret ; \x5f\xc3 (1 found)
```

```
> rp-win-x64.exe -f foo2.bin --raw=x64 --rop=1
0x1: lock syscall ; ret ; \xf0\x0f\x05\xc3 (1 found)
0x2: syscall ; \x0f\x05 (1 found)
0x0: lock syscall ; ret ; \x45\xf0\x0f\x05\xc3 (1 found)
0x2: syscall ; ret ; \x0f\x05\xc3 (1 found)
0x4: ret ; \xc3 (1 found)
```

Does this look good to you?

Cheers

**Comment by 0vercl0k** [comment]
This should be fixed in #30, closing.

Cheers
